server/internal/mcp/impl.go

```diff
@@ -239,6 +247,13 @@ func (s *Service) HandleWellKnownOAuthServerMetadata(w http.ResponseWriter, r *h
 		baseURL = fmt.Sprintf("https://%s", customDomainCtx.Domain)
 	}
 
+	// Pre-resolve external MCP OAuth config for use in switch
+	fullToolset, descErr := mv.DescribeToolset(ctx, s.logger, s.db, mv.ProjectID(toolset.ProjectID), mv.ToolsetSlug(toolset.Slug), &s.toolsetCache)
```

Comment:
Do we need the FULL toolset? DescribeToolset is somewhat expensive

Reply:
https://app.datadoghq.com/logs?query=service%3Agram-server%20%40url.original%3A%2F.well-known%2F%2A&agg_m=count&agg_m_source=base&agg_q=%40url.original&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&flat_group_bys=true&messageDisplay=inline&refresh_mode=sliding&sort_m=count&sort_t=count&storage=hot&stream_sort=desc&top_n=10&top_o=top&viz=timeseries&x_missing=true&from_ts=1765906267874&to_ts=1765992667874&live=true

well-known gets traffic but not a _ton_. All of it is from Polar. Even so I refactored to short-circuit before describe toolset in order to ensure that if there _is_ a latency spike that we're going to avoid it